libqtile/core/manager.py

```diff
@@ -450,6 +460,15 @@ def reconfigure_screens(self, *_: list[Any], **__: dict[Any, Any]) -> None:
     def paint_screen(self, screen: Screen, image_path: str, mode: str | None = None) -> None:
         self.core.painter.paint(screen, image_path, mode)
 
+    def fill_screen(self, screen: Screen, background: ColorType) -> None:
+        with tempfile.NamedTemporaryFile() as temp:
+            surface = cairocffi.ImageSurface(cairocffi.FORMAT_ARGB32, screen.width, screen.height)
+            with cairocffi.Context(surface) as ctx:
+                ctx.set_source_rgba(*rgb(background))
+                ctx.paint()
+            surface.write_to_png(temp.name)
+            self.core.painter.paint(screen, temp.name)
```

Comment:
Can we add a method to get the background surface and do this painting via:

```
ctx.set_source_rgba(0, 0, 0, 0.5)
ctx.rectangle(0, 0, WIDTH, HEIGHT)
ctx.fill()
```
?

it seems a little ugly to me to write a temp file to do this.

Reply:
Sure. I didn't want to create a temporary file. I just did this as it allows us to use the `paint` method. The advantage of that was to use the existing code for updating the relevant atoms in x11.

I was being a bit lazy so let me rework this and see if I can refactor those methods.